### PR TITLE
Clean up some textgen API code

### PR DESCRIPTION
**Pull request by valadaptive**
This PR replaces the API parameters of the form "is_(some AI type)" with fields just indicating which AI type it is. There can only be one AI type at a time, so we don't need multiple boolean params since only one will be true at any given time.

In the future, I may try to refactor more of the logic around textgen APIs if I can figure out how to properly test it. This is a step towards that.

### CODE REVIEW

File: public/script.js [hunk 1]
```diff
@@ -884,21 +879,20 @@ async function getStatus() {
...
+                        textgenerationwebui_settings.type !== textgen_types.MANCER :
+                    false,
             }),
             signal: abortStatusCheck.signal,
         });
 
         const data = await response.json();
 
-        if (main_api == 'textgenerationwebui' && isMancer()) {
+        if (main_api == 'textgenerationwebui' && textgenerationwebui_settings.type === textgen_types.MANCER) {
```
Comment (Cohee1207) [inline public/script.js]: This looks too clunky compared to how it was before, especially in tokenizers.

Comment (valadaptive) [reply public/script.js]: I've tried to shrink things down a bit by renaming the imports and destructuring textgen types when they were getting bulky.

File: public/scripts/openai.js [hunk 2]
```diff
@@ -2463,10 +2459,10 @@ async function getStatusOpen() {
     let data = {
         reverse_proxy: oai_settings.reverse_proxy,
         proxy_password: oai_settings.proxy_password,
-        use_openrouter: oai_settings.chat_completion_source == chat_completion_sources.OPENROUTER,
+        chat_completion_source: oai_settings.chat_completion_source,
```
Comment (Cohee1207) [inline public/scripts/openai.js]: This doesn't really relate to textgen. I'd like to have it as a separate change

File: server.js [hunk 3]
```diff
@@ -177,19 +177,24 @@ function getOverrideHeaders(urlHost) {
...
-    } else if (request.body.use_tabby) {
-        headers = getTabbyHeaders();
-    } else {
-        headers = server ? getOverrideHeaders((new URL(server))?.host) : {};
-    }
+    let headers;
 
-    args.headers = Object.assign(args.headers, headers);
+    switch (request.body.api_type) {
+        case 'mancer':
```
Comment (Cohee1207) [inline server.js]: Might be worth copying the enum to src/constants.js to get rid of strings

File: public/script.js (hunk 1 above)
Comment (Cohee1207) [reply public/script.js]: Okay, that looks better

File: public/scripts/openai.js (hunk 2 above)
Comment (valadaptive) [reply public/scripts/openai.js]: Moved to https://github.com/SillyTavern/SillyTavern/pull/1441

File: server.js (hunk 3 above)
Comment (valadaptive) [reply server.js]: Copied it over

**Comment by Cohee1207** [comment]
Something was wrong when I pulled this related to settings migration. Aphrodite source was set in the settings before, but the request in console was showing 'ooba'

**Comment by valadaptive** [comment]
Found the issue--when I renamed variables to make the checks shorter, I shadowed another variable with the same name. Should be fixed now

**Pull request merged**